sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -47,364 +48,444 @@ const alwaysBeDeletedSubscription = "alwaysbedeletedsubscription";
-            }
-
-            break;
-
-          case EntityType.RULE:
-            try {
-              response = await createEntity(
-                entityType,
-                alwaysBeExistingRule,
-                alwaysBeExistingTopic,
-                alwaysBeExistingSubscription
-              );
-            } catch (err) {
-              response = err;
-            }
-
-            break;
-
-          default:
-            throw new Error("TestError: Unrecognized EntityType");
-        }
+[
+  {
+    entityType: EntityType.QUEUE,
+    alwaysBeExistingEntity: alwaysBeExistingQueue,
+    alwaysBeDeletedEntity: alwaysBeDeletedQueue,
+    parentTopicName: undefined,
+    parentSubscriptionName: undefined
+  },
+  {
+    entityType: EntityType.TOPIC,
+    alwaysBeExistingEntity: alwaysBeExistingTopic,
+    alwaysBeDeletedEntity: alwaysBeDeletedTopic
+  },
+  {
+    entityType: EntityType.SUBSCRIPTION,
+    alwaysBeExistingEntity: alwaysBeExistingSubscription,
+    alwaysBeDeletedEntity: alwaysBeDeletedSubscription,
+    parentTopicName: alwaysBeExistingTopic
+  },
+  {
+    entityType: EntityType.RULE,
+    alwaysBeExistingEntity: alwaysBeExistingRule,
+    alwaysBeDeletedEntity: alwaysBeDeletedRule,
+    parentTopicName: alwaysBeExistingTopic,
+    parentSubscriptionName: alwaysBeExistingSubscription
+  }
+].forEach((testCase) => {
+  describe(`Atom management - Basic CRUD on "${testCase.entityType}" entities #RunInBrowser`, function(): void {
+    it(`Lists on non-existing entities for type ${testCase.entityType} returns empty array successfully`, async () => {
+      let response;
+      try {
+        await deleteEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
 
-        should.equal(response.statusCode, 409, "Error must not be undefined");
-      });
+      response = await listEntities(
+        testCase.entityType,
+        testCase.parentTopicName,
+        testCase.parentSubscriptionName
+      );
 
-      it(`Lists available ${entityType} entities successfully`, async () => {
-        let response;
+      try {
+        await createEntity(
+          testCase.entityType,
+          testCase.alwaysBeExistingEntity,
+          testCase.parentTopicName,
+          testCase.parentSubscriptionName
+        );
+      } catch (err) {
+        log.httpAtomXml("Ignoring clean up step");
+      }
 
-        switch (entityType) {
-          case EntityType.QUEUE:
-          case EntityType.TOPIC:
-            response = await listEntities(entityType);
-            break;
+      should.equal(Array.isArray(response), true, "Result must be any array for list requests");
+      should.equal(
+        response.length,
+        testCase.entityType == EntityType.RULE ? 1 : 0,
+        "Result must be an empty array except for rules"
```

Comment:
Just curious, why is rules non-empty? Does it have a default that's always returned?

Reply:
Yeah, one a default one gets created with name "$Default"